servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -180,7 +180,14 @@ public LogResponse getCommitLog(
       throws NessieNotFoundException {
     int max =
         Math.min(maxRecords != null ? maxRecords : MAX_COMMIT_LOG_ENTRIES, MAX_COMMIT_LOG_ENTRIES);
-    Hash startRef = getHashOrThrow(pageToken != null ? pageToken : ref);
+
+    Ref startRef;
+    if (null == pageToken) {
+      // we should only allow named references when no paging is defined
+      startRef = namedRefWithHashOrThrow(ref).getValue();
+    } else {
+      startRef = getHashOrThrow(pageToken);
```

Comment:
I think, it's fine to accept the "insecure design" (aka: you can pass in any hash and it will give you all the commits) for now until we have both authz and tvs2 in place.
Mind adding a `TODO` here?

Reply:
done